### PR TITLE
Support client version prerelease flag in context and LD targeting

### PR DESCRIPTION
**Pull request by withinfocus**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-14607

## 📔 Objective

Loads the new `Is-Prerelease` client header into context and makes it available to LaunchDarkly.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: src/Core/Context/CurrentContext.cs [hunk 1]
```diff
@@ -97,6 +98,11 @@ public async virtual Task BuildAsync(HttpContext httpContext, GlobalSettings glo
         {
             ClientVersion = cVersion;
         }
+
+        if (httpContext.Request.Headers.TryGetValue("Is-Prerelease", out var clientVersionIsPrerelease))
+        {
+            ClientVersionIsPrerelease = clientVersionIsPrerelease == "true";
```
Comment (trmartin4) [inline src/Core/Context/CurrentContext.cs:104]: @withinfocus are we OK with this being `true` and not an integer value?  I can change the header if that would make this less brittle?

**Review by trmartin4 — APPROVED** [review]
Just one question, but looks good.  Thank you for doing this!

File: src/Core/Context/CurrentContext.cs (hunk 1 above)
Comment (withinfocus) [reply src/Core/Context/CurrentContext.cs:104]: They're all strings in the end, but if you made this 1 or 0 I think it aligns _slightly_ better with some others out there. Lemme know.

Comment (trmartin4) [reply src/Core/Context/CurrentContext.cs:104]: Adjusted this with https://github.com/bitwarden/clients/pull/11909.

**Review by trmartin4 — APPROVED** [review]

**Pull request merged**